control-plane-operator/controllers/hostedcontrolplane/secretproviderclass/secretproviderclass.go

```diff
@@ -34,6 +34,10 @@ func ReconcileManagedAzureSecretProviderClass(secretProviderClass *secretsstorev
 			"objects":                formatSecretProviderClassObject(certName),
 		},
 	}
+
+	if hcp.Spec.Platform.Azure.ObjectEncoding != "" {
+		secretProviderClass.Spec.Parameters["objectEncoding"] = string(hcp.Spec.Platform.Azure.ObjectEncoding)
```

Comment:
I think this has to go in https://github.com/openshift/hypershift/blob/b378274863c22f010a16b6cd7e0d31e46b058557/control-plane-operator/controllers/hostedcontrolplane/secretproviderclass/secretproviderclass.go#L16 

```
array:
  - |
    objectName: <the-name>
    objectType: secret
    objectEncoding: <the-encoding>
```
As par the types in https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/e40b93b7f34e6f8e5f0d278fe7ffa7ad52a69906/pkg/provider/types/types.go#L72

Reply:
I think so too :thinking: , and it is applied per secret :thinking:

Reply:
This also means that if it is added there everyone that calls this method will use that encodingformat, no matter where the code comes from. Is that the intention?